source-academy/cadet-frontend
# Implement collapsible academy navigation bar

### Description



- Implemented collapsible academy navigation bar on desktop
- Academy navigation bar is now accessible at all URLs within the app when user is logged in (previously only accessible at '/academy/*')
- Consolidated all navbar related files under directory '/src/commons/navigationBar'
- Updated CSS for Story Simulator Navlink to prevent overflow when browser is resized

#### Changes to UI (Collapsible academy navbar)
##### Previously:
<img width="1029" alt="image" src="https://user-images.githubusercontent.com/53928333/106014693-ad43f500-60f8-11eb-945c-f7993d914c98.png">

##### Now:
(Opened - default)
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/53928333/106014026-03646880-60f8-11eb-8b87-8efcbc0af97b.png">

(Collapsed)
<img width="1023" alt="image" src="https://user-images.githubusercontent.com/53928333/106014082-1414de80-60f8-11eb-90e2-fe4cf313d403.png">

(Logged out)
<img width="1024" alt="image" src="https://user-images.githubusercontent.com/53928333/106014132-24c55480-60f8-11eb-8659-124cefbe4b9c.png">

#### Changes to UI (Prevent Staff Academy Navbar from overflowing)
##### Previously:
<img width="1030" alt="image" src="https://user-images.githubusercontent.com/53928333/106014499-82f23780-60f8-11eb-8286-6e62ed942d6a.png">

##### Now:
<img width="1029" alt="image" src="https://user-images.githubusercontent.com/53928333/106014402-6c4be080-60f8-11eb-9098-eeea4eaa6949.png">

### Type of change


### How to test


- Test in Playground only configuration (toggle feature should not appear)
- Test in non-Playground only configuration (toggle feature should appear for desktop navbar)
- Toggle button and academy navbar should not render in the mobile breakpoint

### Checklist
